app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,27 @@
+class PostsController < ApplicationController
+  def index
+    @posts = Post.all.order('created_at DESC')
+  end
+
+  def show
+    if params[:id] != "new"
+      @post = Post.find(params[:id])
+    end
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
```

Comment:
At the moment I can write a caption without uploading an image. This is then shown on the feed. When I upload an image without a caption it goes to the feed but it is not posted. I think some alerts would be beneficial, telling the user a post has been created or not giving reasons.